chainerx_cc/chainerx/CMakeLists.txt

```diff
@@ -114,14 +112,29 @@ if(MSVC)
         "${chainerx_srcs}"
         platform/windows.cc
         )
-    add_library(chainerx STATIC empty.cc)
-else()
-    add_library(chainerx SHARED empty.cc)
 endif()
 
+
+add_library(chainerx_base STATIC ${chainerx_srcs})
+
+add_library(chainerx STATIC empty.cc)
+add_library(chainerx_static STATIC empty.cc)
+add_library(chainerx_shared SHARED empty.cc)
+target_link_libraries(chainerx_static
+    PUBLIC
+    -Wl,--whole-archive
```

Comment:
I think these `--whole-archive` options should be added only when building shared library

Reply:
Thanks, it seems it can be removed from `chainerx_static`. (ef47c69baf0e928d9ceecf2cda7f285737a90f0e)
But moving it from `chainerx` to `chainerx_shared` caused undefined symbol errors.